examples/all-clusters-app/all-clusters-common/src/operational-state-delegate-impl.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/EventLogging.h>
+#include <operational-state-delegate-impl.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+using chip::Protocols::InteractionModel::Status;
+
+CHIP_ERROR OperationalStateDelegate::SetOperationalState(const GenericOperationalState & opState)
+{
+    mOperationalState = opState;
+    return CHIP_NO_ERROR;
+}
+
+void OperationalStateDelegate::GetOperationalState(GenericOperationalState & op)
+{
+    op = mOperationalState;
+}
+
+CHIP_ERROR OperationalStateDelegate::SetOperationalError(const GenericOperationalError & opErrState)
+{
+    mOperationalError = opErrState;
+    return CHIP_NO_ERROR;
+}
+
+void OperationalStateDelegate::GetOperationalError(GenericOperationalError & error)
+{
+    error = mOperationalError;
+}
+
+CHIP_ERROR OperationalStateDelegate::GetOperationalStateAtIndex(size_t index, GenericOperationalState & operationalState)
+{
+    if (index > mOperationalStateList.size() - 1)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+    else
+    {
+        operationalState = mOperationalStateList.data()[index];
```

Comment:
```suggestion
        operationalState = mOperationalStateList[index];
```
should work, I would think....